src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -99,6 +99,7 @@ data class PurgeExpiredPayments(val fromCreatedAt: Long, val toCreatedAt: Long)
 
 sealed class PeerEvent
 data class PaymentRequestGenerated(val receivePayment: ReceivePayment, val request: String) : PeerEvent()
+data class PaymentNotReceived(val actions: List<PeerCommand>, val incomingPayment: IncomingPayment?) : PeerEvent()
```

Comment:
Nit: I think the name is confusing, I understood it first as "Payment not yet received".

But most importantly, there is an inconsistency between `PaymentReceived` and `PaymentNotReceived`:
- `PaymentReceived` is for a whole logical payment
- `PaymentNotReceived` will be sent for each rejected payment part

So the two messages don't operate at the same level, and you would have to do some filtering to extract the cases that you are interested in, which I believe is: all parts of a payment have been received, but we are rejecting the payment due to \<reasons\>.

Reply:
The inconsistency is a good point and reveals further issues, I reverted the change. Listening to `ProcessAddResult` is not a good idea.